naturalcrit/homebrewery
# TagInput: Part 2 - Keyboard Navigation

## Description
This portion mostly focuses on improving keyboard navigation of the TagInput elements.  

You should be able to `tab` through the Properties Editor across inputs, jumping to the "new tag" input when you hit the TagInput component.  Further `tab`ing moves you past the TagInput.  

If the "new tag" input is focused, you can use the arrow keys to move back or forwards through the tags/new input, and it *should* trap focus inside the component when using arrow keys-- **it currently does not***.  

You can also `tab`/`shift tab` to move through the tags/new input, and this does *not* trap focus inside the component.

`enter` on a tag opens it for editing or submits it if currently editing, `esc` closes it without saving, `,` submits a change if editing, `tab` moves focus out while keeping it open for editing.

<img width="543" alt="image" src="https://github.com/user-attachments/assets/e6d4e1ac-e428-49de-ac51-ffa795ec5a3d">


## Related Issues or Discussions

- Continues the work on #3735

## QA Instructions, Screenshots, Recordings

Use keyboard to navigate through the TagInput and see if it's intuitive.  If you want to go deeper, I'm sure this could be improved with a closer look at WCAG guidelines, though I did try to follow it pretty closely.

### Reviewer Checklist

*Reviewers, refer to this list when testing features, or suggest new items *
  - [x] `tab` into the component from either before or after the component focuses the 'add tag' input
  - [x] `enter` or `,` in the 'add tag' input submits a new tag
  - [x] `shift tab` from 'add tag' input moves focus through tags starting from end, and escapes the component entirely from the first tag
  - [x] `tab`, when focus is already inside the tag list, moves the focus through tags and escapes the component entirely from the 'add tag' input.
  - [x] `enter` on a tag opens it for editing, and submits it if already open for editing
  - [x] `esc` on edited tag cancels the edit and closes the input
  - [x] `,` on edited tag submits it (like `enter`)
  - [ ] `arrow keys` move focus between tabs but does not escape the component when it reaches the ends
  - [x] still saves tags to the brew as before

<details><summary>Copy this list</summary>

```
  - [ ] `tab` into the component from either before or after the component focuses the 'add tag' input
  - [ ] `enter` or `,` in the 'add tag' input submits a new tag
  - [ ] `shift tab` from 'add tag' input moves focus through tags starting from end, and escapes the component entirely from the first tag
  - [ ] `tab`, when focus is already inside the tag list, moves the focus through tags and escapes the component entirely from the 'add tag' input.
  - [ ] `enter` on a tag opens it for editing, and submits it if already open for editing
  - [ ] `esc` on edited tag cancels the edit and closes the input
  - [ ] `,` on edited tag submits it (like `enter`)
  - [ ] `arrow keys` move focus between tabs but does not escape the component when it reaches the ends
  - [ ] still saves tags to the brew as before

```

</details>
